repos/system_upgrade/common/libraries/tests/test_rhsm.py

```diff
@@ -119,3 +209,169 @@ def test_inhibit_on_duplicate_repos_no_dups(monkeypatch):
+    """Tests whether the library obtains the SCA information correctly from a non-SCA system."""
+    context_mocked.add_mocked_command_call_with_stdout(RHSM_STATUS, RHSM_STATUS_OUTPUT_NONSATELLITE)
+
+    is_sca = rhsm.is_manifest_sca(context_mocked)
+    assert not is_sca, 'SCA was detected on a non-SCA system.'
+
+    context_mocked = IsolatedActionsMocked()  # Do not share context with previous is_manifest_sca call
+    context_mocked.add_mocked_command_call_with_stdout(RHSM_STATUS, RHSM_STATUS_OUTPUT_SATELLITE)
+
+    is_sca = rhsm.is_manifest_sca(context_mocked)
+    assert not is_sca, 'SCA was detected on a non-SCA system.'
+
+
+def test_is_manifest_sca_on_sca_system(monkeypatch, actor_mocked, context_mocked):
+    """Tests whether the library obtains the SCA information from SCA system correctly."""
+    context_mocked.add_mocked_command_call_with_stdout(RHSM_STATUS, RHSM_STATUS_OUTPUT_SCA)
+
+    is_sca = rhsm.is_manifest_sca(context_mocked)
+    assert is_sca, 'Failed to detected SCA on a SCA system.'
+
+
+def test_get_enabled_repo_ids(monkeypatch, actor_mocked, context_mocked):
+    """Tests whether the library retrieves correct information about enabled repositories."""
+    # Prepare the (realistic) RHSM output
+    rhsm_list_enabled_output = '''
+    +----------------------------------------------------------+
+       Available Repositories in /etc/yum.repos.d/redhat.repo
+    +----------------------------------------------------------+
+    '''
+
+    for enabled_repository in RHSM_ENABLED_REPOS:
+        rhsm_output_fragment = 'Repo ID: {0}\n'.format(enabled_repository.id)
+        rhsm_output_fragment += 'Repo Name: {0}\n'.format(enabled_repository.name)
+        rhsm_output_fragment += 'Repo URL: {0}\n'.format(enabled_repository.url)
+        rhsm_output_fragment += 'Enabled: {0}\n'.format(enabled_repository.enabled)
+        rhsm_output_fragment += '\n'
+        rhsm_list_enabled_output += rhsm_output_fragment
+
+    context_mocked.add_mocked_command_call_with_stdout(RHSM_LIST_ENABLED_REPOS, rhsm_list_enabled_output)
+
+    enabled_repo_ids = rhsm.get_enabled_repo_ids(context_mocked)
+
+    fail_description = 'Failed to detected enabled repositories on the system.'
+    assert len(enabled_repo_ids) == 3, fail_description
+
+    fail_description = 'Failed to retrieve repository ID provided in the RHSM output.'
+    for enabled_repository in RHSM_ENABLED_REPOS:
+        assert enabled_repository.id in enabled_repo_ids, fail_description
+
+
+def test_get_existing_product_certificates(monkeypatch, actor_mocked, context_mocked):
+    """Verifies that the library is able to correctly retrieve existing product certificates."""
+
+    def mocked_isdir(path):
+        # Only return True for these two specific paths specific to this test
+        if path == '/etc/pki/product-default':
+            return True
+        if path == '/etc/pki/product':
+            return False
```

Comment:
Comment says it should return `Tru`, but it returns `False` in the second case. Shouldn't it be rather (?):
```suggestion
        # Only return True for these two specific paths specific to this test
        if path in ['/etc/pki/product-default', '/etc/pki/product']:
            return True
```

Or fix the comment. Regarding the stuff in the bottom, I guess you wanted a different comment. However the setup like /etc/pki/product is file seems not reflecting the real world. But I am ok with the test too. However in such a case, I would expect even the case when both are dirs as in the real world, as this is the most common use case. Real scenarious are more valuable for us to test than very unexpected scenarious. Regarding amount of cases covered in this test function, a second one could be much more simple (e.g. expected two certs are discovered, one in product, one in product-default, no other problems).

Reply:
Hmm... the entire `test_get_existing_product_certificates` is riddled with unnecessary logic. I guess I tried to be smart and test whether the library can cope with some edge case scenarios. Going to rewrite the test to contain a real world scenario that will be tested thoroughly.

Reply:
Tests for the `get_existing_product_certificates` now consist of two tests - one thoroughly testing common scenario, one testing edge case when one of the cert directories is missing.